### PR TITLE
added redux and hook up navigation service

### PR DESCRIPTION
**Pull request by mrayor**
- Hooked up navigation service to 

- setup redux and redux-saga

### CODE REVIEW

File: src/store/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { createStore, applyMiddleware, compose } from 'redux';
+import createSagaMiddleware from 'redux-saga';
+import rootReducer from './rootReducer';
+// import rootSaga from "./rootSaga";
```
Comment (parkerthegeniuschild) [inline src/store/index.js]: take out commented code

File: src/store/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+let composeEnhancers = compose;
+
+composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+
+const store = createStore(
+    rootReducer,
+    initialState,
+    composeEnhancers(applyMiddleware(...middleware))
+);
+// sagaMiddleware.run(rootSaga);
```
Comment (parkerthegeniuschild) [inline src/store/index.js]: take out commented code

File: App.jsx [hunk 3]
```diff
@@ -13,6 +16,13 @@ const AppNavigator = createSwitchNavigator({
...
   render() {
-    return <AppContainer />;
+    return (
+      <Provider store={store}>
+        <AppContainer ref={navigatorRef => {
+          NavigationService.setTopLevelNavigator(navigatorRef)
+        }}/>
+
+      </Provider>
+    );
```
Comment (enyeInc) [inline App.jsx]: fix the indents on this file

**Comment by ekediala** [comment]
Merge away. It's all good.

File: App.jsx (hunk 3 above)
Comment (mrayor) [reply App.jsx]: done

**Pull request merged**